airbyte-cdk/java/airbyte-cdk/core/src/main/java/io/airbyte/cdk/integrations/config/AirbyteSourceConfig.java

```diff
@@ -0,0 +1,70 @@
+package io.airbyte.cdk.db;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import io.airbyte.commons.io.IOs;
+import io.airbyte.commons.json.Jsons;
+import java.nio.file.Path;
+import java.util.Collections;
+import java.util.Map;
+
+public class AirbyteSourceConfig extends AirbyteConfig<AirbyteSourceConfig>{
+  public static class SourceConfigBuilder extends AirbyteConfigBuilder<AirbyteSourceConfig> {
+    public SourceConfigBuilder(AirbyteSourceConfig sourceConfig) {
+      super(sourceConfig);
+    }
+
+    public AirbyteSourceConfig build() {
+      return new AirbyteSourceConfig(jsonConfig);
+    }
+
+    public SourceConfigBuilder with(String key, String value) {
+      jsonConfig.put(key, value);
+      return this;
+    }
+
+    public SourceConfigBuilder with(String key, boolean value) {
+      jsonConfig.put(key, value);
+      return this;
+    }
+  }
+
+  private AirbyteSourceConfig(JsonNode jsonConfig) {
+    super(jsonConfig);
+  }
+
+  public static AirbyteSourceConfig fromPath(final Path path) {
+    return new AirbyteSourceConfig(Jsons.deserialize(IOs.readFile(path)));
+  }
+
+  public static AirbyteSourceConfig fromJsonNode(final JsonNode jsonNode) {
+    return new AirbyteSourceConfig(jsonNode);
+  }
+
+  public static AirbyteSourceConfig fromJsonString(final String jsonString) {
+    return new AirbyteSourceConfig(Jsons.deserialize(jsonString));
+  }
+
+  public SourceConfigBuilder cloneBuilder() {
```

Comment:
Alternatively, and perhaps preferably, have the builder methods return a deep copy of themselves instead of `this`, that way you never have to worry about state ownership ever again.